src/ArmTemplates/Commands/Executors/ExtractorExecutor.cs

```diff
@@ -708,6 +708,12 @@ public async Task<Template<GroupTemplateResources>> GenerateGroupsTemplateAsync(
         /// <returns>generated identity provider template</returns>
         public async Task<Template<IdentityProviderResources>> GenerateIdentityProviderTemplateAsync(string baseFilesGenerationDirectory)
         {
+            if (this.extractorParameters.ExtractIdentityProviders == false)
+            {
+                this.logger.LogInformation($"Skipping identityProvider extraction due to the setting parameter");
+                return null;
```

Comment:
instead of returning `null`, you can create an empty `IdentityProviderResources` instance. You can even make it not using a constructor, but by having a predefined empty instance already.

Just for reference - [string.Empty](https://source.dot.net/#System.Private.CoreLib/String.cs,38) has the same approach.
This will help not having troubles with NRE later, since we will be using `HasResources()` method later also, and you will not need to add nullable checker `?.` to all of usages.

This is not that important, but I think later it will be a better approach - you will not ever care about NREs

Reply:
I was thinking about that, but decided to go with null, but yes in future we could face some NRE when not properly handle identity providers. 
Thank you for the suggestion! updated accordingly.